### PR TITLE
bump: slf4j 2.0.16, logback 1.5.7

### PR DESCRIPTION
**Pull request by sebastian-alfers**
Reverts the reflection based workaround introduced in https://github.com/akka/akka/pull/31825.

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

**Comment by johanandren** [comment]
There's some docs that need updating as well: https://doc.akka.io/docs/akka/current/typed/logging.html#slf4j-api-compatibility

**Comment by sebastian-alfers** [comment]
> There's some docs that need updating as well: https://doc.akka.io/docs/akka/current/typed/logging.html#slf4j-api-compatibility

Good point. I added a small correction to that part. But maybe we can even drop the entire "SLF4J API compatibility" section?

File: akka-docs/src/main/paradox/typed/logging.md [hunk 1]
```diff
@@ -173,9 +173,9 @@ in the `ActorContext`.
 
 ## SLF4J API compatibility
 
-The SLF4J API broke binary compatibility between versions 1.7 and 2.0. Akka depends on 1.7 but supports either versions. 
+Since Akka 2.10.0, only SLF4j version `2.0` is supported. 
 
-It is however not possible to mix a logger backend supporting one version with SLF4J API of other version, that will lead
+It is not possible to mix a logger backend supporting one version with SLF4J API of other version, that will lead
```
Comment (johanandren) [inline akka-docs/src/main/paradox/typed/logging.md:178]: Or possibly, more explicit:
```suggestion
It is not possible to mix a logger backend supporting one version with SLF4J API of older versions, that will lead
```

**Review by johanandren — APPROVED** [review]

File: akka-docs/src/main/paradox/typed/logging.md [hunk 2]
```diff
@@ -173,9 +173,9 @@ in the `ActorContext`.
 
 ## SLF4J API compatibility
 
-The SLF4J API broke binary compatibility between versions 1.7 and 2.0. Akka depends on 1.7 but supports either versions. 
+Since Akka 2.10.0, only SLF4j version `2.0` is supported. 
 
-It is however not possible to mix a logger backend supporting one version with SLF4J API of other version, that will lead
+It is not possible to mix a logger backend supporting one version with SLF4J API of older versions, that will lead
 to no logging to output like this:
```
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/logging.md:179]: Is below log message obsolete now? I don't think we want to encourage downgrading slf4j as is explained in below workaround.

**Review by patriknw — COMMENTED** [review]
Can you also start a new migration page where we mention this dependency update.
https://github.com/akka/akka/tree/main/akka-docs/src/main/paradox/project

File: akka-docs/src/main/paradox/typed/logging.md (hunk 2 above)
Comment (sebastian-alfers) [reply akka-docs/src/main/paradox/typed/logging.md:179]: Yes, lets remove all the "working around this" docs.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
Refs https://github.com/akka/akka/issues/32364

**Pull request merged**